// signaling-server/dist/src/index.js
"use strict";

// export interface Config {
//   UseFrontend: boolean;
//   UseMatchmaker: boolean;
//   UseHTTPS: boolean;
//   UseAuthentication: boolean;
//   LogToFile: boolean;
//   LogVerbose: boolean;
//   HomepageFile: string;
//   AdditionalRoutes: Map<string, string>;
//   EnableWebserver: boolean;
//   MatchmakerAddress: string;
//   MatchmakerPort: number;
//   PublicIp: string;
//   HttpPort: number;
//   HttpsPort: number;
//   StreamerPort: number;
//   SFUPort: number;
//   MaxPlayerCount: number;
// }

const defaultConfig = {
  UseFrontend: false,
  UseMatchmaker: false,
  UseHTTPS: false,
  UseAuthentication: false,
  LogToFile: true,
  LogVerbose: true,
  HomepageFile: 'player.html',
  AdditionalRoutes: new Map(),
  EnableWebserver: true,
  MatchmakerAddress: "",
  MatchmakerPort: 9999,
  PublicIp: "localhost",
  HttpPort: 80,
  HttpsPort: 443,
  StreamerPort: 8888,
  SFUPort: 8889,
  MaxPlayerCount: -1
};
const cirrus = require('./cirrus.js');
const WebServer = (config = {}) => {
  config = {
    ...defaultConfig,
    ...config
  };
  cirrus(config);
};

// test
// WebServer()

module.exports = WebServer;
